graylog2-server/src/main/java/org/graylog2/rest/resources/search/SearchResource.java

```diff
@@ -58,22 +61,32 @@
 import javax.ws.rs.core.Response;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 import java.util.Locale;
+import java.util.Optional;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Strings.isNullOrEmpty;
 
 public abstract class SearchResource extends RestResource {
     private static final Logger LOG = LoggerFactory.getLogger(SearchResource.class);
+    private static final Pattern filterStreamIdPattern = Pattern.compile("^(.+[^\\p{Alnum}])?streams:([\\p{XDigit}]+)");
 
     protected final Searches searches;
     private final ClusterConfigService clusterConfigService;
+    private final DecoratorProcessor decoratorProcessor;
 
     @Inject
-    public SearchResource(Searches searches, ClusterConfigService clusterConfigService) {
+    public SearchResource(Searches searches,
+                          ClusterConfigService clusterConfigService,
+                          DecoratorProcessor decoratorProcessor) {
```

Comment:
Did that and opened a PR: https://github.com/Graylog2/graylog-plugin-map-widget/pull/14